app/models/user/reputation_tokens/arbitrary_token.rb

```diff
@@ -0,0 +1,22 @@
+class User::ReputationTokens::ArbitraryToken < User::ReputationToken
+  params :arbitrary_value, :arbitrary_reason
+  category :other
+
+  before_validation on: :create do
+    self.earned_on = Time.current unless earned_on
+    self.reason = arbitrary_reason unless reason
+  end
+
+  def guard_params
+    SecureRandom.uuid
+  end
+
+  def i18n_params
+    { reason: }
+  end
+
+  protected
+  def determine_value
+    arbitrary_value
+  end
```

Comment:
Want to go all posh Ruby 3?
```suggestion
  def determine_value = arbitrary_value
```

Reply:
😮

Reply:
But: yes!